pkg/models/commandvalue.go

```diff
@@ -167,14 +181,26 @@ func NewFloat64Value(ro *models.ResourceOperation, origin int64, value float64)
 //NewCommandValue create a CommandValue according to the Type supplied
 func NewCommandValue(ro *models.ResourceOperation, origin int64, value interface{}, t ValueType) (cv *CommandValue, err error) {
 	cv = &CommandValue{RO: ro, Origin: origin, Type: t}
-	if t != String {
+	if t == Binary {
+		// assign cv.binValue
+		err = encodeBinaryValue(cv, value)
+	} else if t != String {
 		err = encodeValue(cv, value)
 	} else {
 		cv.stringValue = value.(string)
 	}
 	return
 }
 
+// NewBinaryValue creates a CommandValue with binary payload and enforces the memory limit for event readings.
+func NewBinaryValue(ro *models.ResourceOperation, origin int64, value []byte) (cv *CommandValue, err error) {
+	if binary.Size(value) > MaxBinaryBytes {
```

Comment:
We return an error here if caller of NewBinaryValue adds a payload > MaxBinaryBytes (1MB)